cmd/minishift/cmd/openshift/service.go

```diff
@@ -50,25 +53,90 @@ var serviceCmd = &cobra.Command{
 			atexit.ExitWithMessage(1, "You must specify the name of the service.")
 		}
 
+		host, err := api.Load(constants.MachineName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+
+		util.ExitIfNotRunning(host.Driver, constants.MachineName)
+
+		ip, err := host.Driver.GetIP()
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error getting IP: %s", err.Error()))
+		}
+
 		service := args[0]
 
-		url, err := openshift.GetServiceSpec(service, namespace, https)
+		serviceSpec, err := openshift.GetServiceSpec(service, namespace)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
 
-		if urlMode {
-			fmt.Fprintln(os.Stdout, url)
+		if url {
+			stdOutURL(serviceSpec)
+		} else if inbrowser {
+			openInBrowser(serviceSpec, https, ip)
 		} else {
-			fmt.Fprintln(os.Stdout, "Opening the service "+namespace+"/"+service+" in the default browser...")
-			browser.OpenURL(url)
+			printToStdOut(serviceSpec, ip)
 		}
 	},
 }
 
 func init() {
 	serviceCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "The namespace of the service.")
-	serviceCmd.Flags().BoolVar(&urlMode, "url", false, "Access the service in the command-line console instead of in the default browser.")
+	serviceCmd.Flags().BoolVar(&inbrowser, "inbrowser", false, "Access the service in the default browser.")
+	serviceCmd.Flags().BoolVar(&url, "url", false, "Access the service url in the command-line console.")
 	serviceCmd.Flags().BoolVar(&https, "https", false, "Access the service with HTTPS instead of HTTP.")
 	OpenShiftCmd.AddCommand(serviceCmd)
 }
+
+func openInBrowser(serviceSpec openshift.ServiceSpec, https bool, ip string) {
+	if serviceSpec.URL != nil {
+		fmt.Fprintln(os.Stdout, "Opening the service route "+serviceSpec.URL[0]+" in the default browser...")
+		browser.OpenURL(serviceSpec.URL[0])
+
+	} else if serviceSpec.NodePort != "" {
+		nodePortURL := fmt.Sprintf("%s:%s", ip, serviceSpec.NodePort)
+		urlScheme := "http://"
+		if https {
+			urlScheme = "https://"
+		}
+		fmt.Fprintln(os.Stdout, "Opening the service nodeport "+urlScheme+nodePortURL+" in the default browser...")
```

Comment:
"Opening NodePort "+urlScheme+nodePortURL+" in the default browser..."

I don't think 'service is needed here. Also, everywhere I see node ports mentioned, it is in the format "NodePort" never "nodeport".